### PR TITLE
Remove Netty from Spark Client Shaded Jar

### PR DESCRIPTION
**Pull request by pan3793**
### What changes were proposed in this pull request?

- Align Netty version w/ Spark
- Remove Netty from Spark Client Shaded Jar

### Why are the changes needed?

Currently, Celeborn Spark Client Shaded Jar ships the relocated netty classes, but since there are native libs inside Netty, and it's not easy(IMO) to relocate native libs, then it still has conflict potentials.

image

### How was this patch tested?

/cc @related-reviewer

/assign @main-reviewer

### CODE REVIEW

**Comment by pan3793** [comment]
I'm going to fix the compile error after #891 get merged.

**Comment by FMX** [comment]
Let us shade netty and rename the native libraries to keep the netty version under control because I have seen some performance gain and stability improvement after using the latest netty. And there will be more engines to support, a stable netty will be helpful for us to profile the performance.
So I think this might not be a good idea.

File: common/src/main/java/org/apache/celeborn/common/network/util/NettyUtils.java [hunk 1]
```diff
@@ -106,17 +106,32 @@ public static String getRemoteAddress(Channel channel) {
    * released by the executor thread rather than the event loop thread. Those thread-local caches
    * actually delay the recycling of buffers, leading to larger memory usage.
    */
+  @SuppressWarnings("deprecation")
   public static PooledByteBufAllocator createPooledByteBufAllocator(
       boolean allowDirectBufs, boolean allowCache, int numCores) {
     if (numCores == 0) {
       numCores = Runtime.getRuntime().availableProcessors();
     }
+    // After upgrade to Netty 4.1.75, there are 2 behavior changes of this method:
```
Comment (pan3793) [inline common/src/main/java/org/apache/celeborn/common/network/util/NettyUtils.java:115]: is this related to the perf gain you mentioned? @FMX

**Comment by pan3793** [comment]
There is an example how to relocate Netty native libs in Gradle
https://github.com/grpc/grpc-java/blob/master/netty/shaded/build.gradle

File: common/src/main/java/org/apache/celeborn/common/network/util/NettyUtils.java (hunk 1 above)
Comment (FMX) [reply common/src/main/java/org/apache/celeborn/common/network/util/NettyUtils.java:115]: Yes, and there are some other reasons to perf gain.

**Comment by FMX** [comment]
> There is an example how to relocate Netty native libs in Gradle https://github.com/grpc/grpc-java/blob/master/netty/shaded/build.gradle

Message received, I think maven can do it as well. I'll try it.

**Comment by pan3793** [comment]
I found https://github.com/apache/flink-shaded done the job, thanks for the explanation, I agree with you as long as we can fully shade the Netty native libs, then close this PR.
